behavior:
  head:
    # Name of the head_behaviors ros-node
    rosnode: head_behavior

    # Sane default values for some modules
    defaults:
      head_mode: 0    # Ball mode

    # Search patterns in degrees, [pan, tilt]
    search_pattern:
      - [0, -15]
      - [0, -65]
      - [63, -57]
      - [-63, -57]
      - [-57, -30]
      - [57, -30]
      - [30, -7]
      - [-30, -7]

    # These values describe the minimal required delta between current joint states and target joint states in degrees
    # to reduce unnecessary movement due to noise in the detection of the BallRelative.
    ball_tracking_min_pan_delta: 8
    ball_tracking_min_tilt_delta: 5

    # Positions for static head modes
    look_down_position: [0, -65]
    look_forward_position: [0, -7]
    look_up_position: [0, 12]

    # Threshold (in degrees) when a head position is reached and
    # the next position will be triggered
    position_reached_threshold: 5

    # After `ball_lost_time` seconds, the ball is considered lost and will be searched
    ball_lost_time: 1
    post_lost_time: 1
